docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,761 @@
+
+:create-app-artifact-id: quarkus-auth0
+:create-app-extensions: resteasy-reactive,oidc
+include::{includes}/devtools/create-app.adoc[]
+
+Create the application workspace and import it into your favourite IDE, note a `GreetingResource` JAX-RS resource that will need to be secured:
+
+[source,java]
+----
+package org.acme;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello from RestEasy Reactive";
+    }
+}
+----
+
+Update the Quarkus application code so that only authenticated users can access the `hello()` method and update it to return the current username instead, as shown in the following example:
+
+[source,java]
+----
+package org.acme;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.oidc.IdToken;
+import io.quarkus.security.Authenticated;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @Inject
+    @IdToken                                        <1>
+    JsonWebToken idToken;
+	
+    @GET
+    @Authenticated                                  <2>
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello, " + idToken.getName();
+    }
+}
+----
+<1> The injected `JsonWebToken` (JWT) token has an `@IdToken` qualifier.
+`IdToken` represents the user authentication part of the OIDC authorization code flow mechanism.
+<2> The `io.quarkus.security.Authenticated` annotation is added to the `hello()` method, which means that only authenticated users can access it.
+
+[NOTE]
+====
+The access token acquired during the authorization code flow, alongside the ID token, is not used directly by the endpoint but is only used to access downstream services on behalf of the currently authenticated user. 
+More to come on the topic of "access tokens", later in this tutorial.
+
+Configure OIDC in `application.properties` with the properties from the created `Auth0` application:
+
+[source,configuration]
+----
+# Make sure the application domain is prefixed with 'https://'
+quarkus.oidc.auth-server-url=https://dev-3ve0cgn7.us.auth0.com
+quarkus.oidc.application-type=web-app
+quarkus.oidc.client-id=sKQu1dXjHB6r0sra0Y1YCqBZKWXqCkly
+quarkus.oidc.credentials.secret=${client-secret}
+----
+
+Here, you have configured Quarkus to use the created `Auth0` application's domain, client id and secret. `quarkus.oidc.application-type=web-app` is an indication to Quarkus that the OIDC Authorization code flow must be used (we'll consider the alternatives later in this tutorial).
```

Comment:
```suggestion
In completing this step, you have just configured Quarkus to use the domain, client ID, and secret of the `Auth0` application that you created earlier. 
Setting the property `quarkus.oidc.application-type=web-app` instructs Quarkus to use the OIDC authorization code flow but there are also other methods, which are discussed later on in the tutorial.
```

Reply:
@michelle-purcell `In completing this step` definitely reads better than `Here` :-)